src/output.cpp

```diff
@@ -0,0 +1,736 @@
+      << mean << " +/- " << t_n1 * stdev << '\n';
+  } else {
+    if (mpi::master) warning("Could not compute uncertainties -- only one "
+      "active batch simulated!");
+
+    if (settings::run_mode == RUN_MODE_EIGENVALUE) {
+      std::cout << " k-effective (Collision)    = "
+        << gt(K_COLLISION, RESULT_SUM) / n << '\n';
+      std::cout << " k-effective (Track-length) = "
+        << gt(K_TRACKLENGTH, RESULT_SUM) / n << '\n';
+      std::cout << " k-effective (Absorption)   = "
+        << gt(K_ABSORPTION, RESULT_SUM) / n << '\n';
+    }
+    std::cout << " Leakage Fraction           = "
+      << gt(LEAKAGE, RESULT_SUM) / n << '\n';
+  }
+  std::cout << '\n';
+
+  // Restore state of cout
+  std::cout.flags(f);
+}
+
+//==============================================================================
+
+const std::unordered_map<int, const char*> score_names = {
+  {SCORE_FLUX,               "Flux"},
+  {SCORE_TOTAL,              "Total Reaction Rate"},
+  {SCORE_SCATTER,            "Scattering Rate"},
+  {SCORE_NU_SCATTER,         "Scattering Production Rate"},
+  {SCORE_ABSORPTION,         "Absorption Rate"},
+  {SCORE_FISSION,            "Fission Rate"},
+  {SCORE_NU_FISSION,         "Nu-Fission Rate"},
+  {SCORE_KAPPA_FISSION,      "Kappa-Fission Rate"},
+  {SCORE_EVENTS,             "Events"},
+  {SCORE_DECAY_RATE,         "Decay Rate"},
+  {SCORE_DELAYED_NU_FISSION, "Delayed-Nu-Fission Rate"},
+  {SCORE_PROMPT_NU_FISSION,  "Prompt-Nu-Fission Rate"},
+  {SCORE_INVERSE_VELOCITY,   "Flux-Weighted Inverse Velocity"},
+  {SCORE_FISS_Q_PROMPT,      "Prompt fission power"},
+  {SCORE_FISS_Q_RECOV,       "Recoverable fission power"},
+  {SCORE_CURRENT,            "Current"},
+};
+
+//! Create an ASCII output file showing all tally results.
+
+void
+write_tallies()
+{
+  if (model::tallies.empty()) return;
+
+  // Open the tallies.out file.
+  std::ofstream tallies_out;
+  tallies_out.open("tallies.out", std::ios::out | std::ios::trunc);
+  tallies_out << std::setprecision(6);
+
+  // Loop over each tally.
+  for (auto i_tally = 0; i_tally < model::tallies.size(); ++i_tally) {
+    const auto& tally {*model::tallies[i_tally]};
+
+    // Write header block.
+    std::string tally_header("TALLY " + std::to_string(tally.id_));
+    if (!tally.name_.empty()) tally_header += ": " + tally.name_;
+    tallies_out << header(tally_header) << "\n\n";
+
+    if (!tally.writable_) {
+      tallies_out << " Internal\n\n";
+      continue;
+    }
+
+    // Calculate t-value for confidence intervals
+    double t_value = 1;
+    if (settings::confidence_intervals) {
+      auto alpha = 1 - CONFIDENCE_LEVEL;
+      t_value = t_percentile(1 - alpha*0.5, tally.n_realizations_ - 1);
+    }
+
+    // Write derivative information.
+    if (tally.deriv_ != C_NONE) {
+      const auto& deriv {model::tally_derivs[tally.deriv_]};
+      switch (deriv.variable) {
+      case DIFF_DENSITY:
+        tallies_out << " Density derivative  Material "
```

Comment:
Nit: Not sure if this was intentional, but there is a extra space between "derivative" and "Material", both here and on lines 667 and 672.